xbmc/Application.cpp

```diff
@@ -2453,7 +2445,7 @@ void CApplication::OnApplicationMessage(ThreadMessage* pMsg)
     break;
 
   case TMSG_SETVIDEORESOLUTION:
-    g_graphicsContext.SetVideoResolution(static_cast<RESOLUTION>(pMsg->param1), pMsg->param2 == 1);
+    g_graphicsContext.SetVideoResolution(static_cast<RESOLUTION>(pMsg->param1), pMsg->param2 & 1, pMsg->param2 & 2);
```

Comment:
it is not in the responsibility of the caller how this call will be handled. 3rd parameter should not be required